src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -267,19 +267,29 @@ export class BrowserRenderer {
       this.trySetSelectValueFromOptionElement(newDomElementRaw);
     } else if (deferredValuePropname in newDomElementRaw) {
       // Situation 2
-      const deferredValue: string | null = newDomElementRaw[deferredValuePropname];
-      setDeferredElementValue(newDomElementRaw, deferredValue);
+      setDeferredElementValue(newDomElementRaw, newDomElementRaw[deferredValuePropname]);
     }
   }
 
   private trySetSelectValueFromOptionElement(optionElement: HTMLOptionElement) {
     const selectElem = this.findClosestAncestorSelectElement(optionElement);
-    if (selectElem && (deferredValuePropname in selectElem) && selectElem[deferredValuePropname] === optionElement.value) {
-      setDeferredElementValue(selectElem, optionElement.value);
-      delete selectElem[deferredValuePropname];
-      return true;
+
+    if (!selectElem || !(deferredValuePropname in selectElem)) {
+      return false;
+    }
+
+    if (selectElem.type === 'multiple-select') {
```

Comment:
Looks like this should be `select-multiple` as everywhere else? Could also share this helper method with the other files  [`isMultipleSelectInput`](https://github.com/dotnet/aspnetcore/pull/33950/files#diff-58576c4c052d6fbfb851ec2f4b3bf740a2e318f0a9aeec9bc7aaf7a8acffe2eaR252).

```suggestion
    if (selectElem.type === 'select-multiple') {
```